src/sqlancer/oceanbase/oracle/OceanBaseNoRECOracle.java

```diff
@@ -0,0 +1,204 @@
+                               List<OceanBaseExpression> groupBys) throws SQLException {
+        OceanBaseSelect select = new OceanBaseSelect();
+        select.setGroupByClause(groupBys);
+        OceanBaseExpression expr = getTrueExpr(randomWhereCondition);
+
+        OceanBaseText asText = new OceanBaseText(expr, " as count", false);
+        select.setFetchColumns(Arrays.asList(asText));
+        select.setFromList(tableList);
+        select.setSelectType(OceanBaseSelect.SelectType.ALL);
+        int secondCount = 0;
+
+        unoptimizedQueryString = "SELECT SUM(count) FROM (" + OceanBaseVisitor.asString(select) + ") as asdf";
+        SQLQueryAdapter q = new SQLQueryAdapter(unoptimizedQueryString, errors);
+        SQLancerResultSet rs;
+        if (options.logEachSelect()) {
+            logger.writeCurrent(unoptimizedQueryString);
+        }
+        try {
+            rs = q.executeAndGet(state);
+        } catch (Exception e) {
+            throw new AssertionError(optimizedQueryString, e);
+        }
+        if (rs == null) {
+            return -1;
+        }
+        if (rs.next()) {
+            secondCount += rs.getLong(1);
+        }
+        rs.close();
+        return secondCount;
+    }
+
+    private int getFirstQueryCount(List<OceanBaseExpression> tableList, OceanBaseExpression randomWhereCondition,
+            List<OceanBaseExpression> groupBys) throws SQLException {
+        OceanBaseSelect select = new OceanBaseSelect();
+        select.setGroupByClause(groupBys);
+        // SELECT COUNT(t1.c3) FROM t1 WHERE (- (t1.c2));
+        // SELECT SUM(count) FROM (SELECT ((- (t1.c2)) IS TRUE) as count FROM t1);;
+        OceanBaseAggregate aggr = new OceanBaseAggregate(
+                new OceanBaseColumnName(new OceanBaseSchema.OceanBaseColumn("*", OceanBaseSchema.OceanBaseDataType.INT, false, 0, false)),
+                OceanBaseAggregate.OceanBaseAggregateFunction.COUNT);
+        select.setFetchColumns(Arrays.asList(aggr));
+        select.setFromList(tableList);
+        select.setWhereClause(randomWhereCondition);
+        select.setSelectType(OceanBaseSelect.SelectType.ALL);
+        int firstCount = 0;
+        optimizedQueryString = OceanBaseVisitor.asString(select);
+        SQLQueryAdapter q = new SQLQueryAdapter(optimizedQueryString, errors);
+        SQLancerResultSet rs;
+        if (options.logEachSelect()) {
+            logger.writeCurrent(optimizedQueryString);
+        }
+        try {
+            rs = q.executeAndGet(state);
+        } catch (Exception e) {
+            throw new AssertionError(firstQueryString, e);
+        }
+        if (rs == null) {
+            return -1;
+        }
+        if (rs.next()) {
+            firstCount += rs.getLong(1);
+        }
+        rs.close();
+        return firstCount;
+    }
+  
+    private OceanBaseExpression getTrueExpr(OceanBaseExpression randomWhereCondition){
```

Comment:
Perhaps you can add a short comment explaining how this differs from the standard NoREC version?